### DISCUSSION TITLE
Reopen an accidentally merged PR

### DISCUSSION THREAD

**Discussion by lrozenblyum** [community/community#37189] [Pull Requests]
Let's have a scenario when a large PR (200+ files) to an unprotected branch is being reviewed by multiple reviewers.

One of them accidentally merges it while there exist still open conversations and other reviewers haven't finished the review.

In this case it would be really great to have a possibility to reopen such pull request.

What is possible now:
* restore the merged branch if it was auto-deleted
* reset HEAD of the target branch to the commit before the merge commit.

However what is not possible now (tried various solutions from https://stackoverflow.com/questions/12674304/github-reopening-a-merged-pull-request were not working):

reopen the PR itself, instead we need to create a new 1.

Why is it a problem:

* 'Viewed' status of files in the PR is lost - reviewers who actively using these checkboxes to track their review progress need to start from scratch.
* The conversations exist in the closed PR - and need to be checked separately, they're not connected to the newly created PR.

So please consider a possibility to reopen a PR.

**Comment by klaudiusmiskinis** [comment]
But you are able to **revert** the changes, after reverting now you can go to the branch and from the last commit create a new branch or just stay in the same branch and work on it and create a new Pull Request with the exact same moment it was before accepting the merge.

**Reply by lrozenblyum** [reply]
@klaudiusmiskinis yes it's possible, but as I mentioned above in this case we lose the reviewers 'Viewed' files progress and conversations are inconvenient to reach.

**Comment by ossanna16** [comment]
Hi there @hwansday and welcome to our community! Thank you for your suggestion 🙂

**Comment by technillogue** [comment]
also interested in this, it would be great to keep conversation history

**Comment by MaskRay** [comment]
Thanks for creating this feature request.

The description is about "accidentally merges" but the supporting arguments well apply to other PRs that are identified as faulty and therefore reverted.

**Comment by ocornut** [comment]
Additionally GitHub has a bug with branch rename **which can erroneously close some pull requests** (which I have reported as ticket 2006174 and was acknowledged, posted a recap at https://github.com/ocornut/imgui/issues/6141#issuecomment-1872223896). Presently I calculated that renaming my "master" branch to "main" would erroneously close about 10 PR which I won't be able to reopen. I track open PR and it is unacceptable to me to lose that information.

If I could reopen a closed PR I would be able to rename my branch.

**Comment by ilchenkoArtem** [comment]
Any update?

**Comment by kurtbahartr** [comment]
Apparently no updates and a friend of mine got struck with an accidental merge too. She accidentally pushed the `dev` branch to `main` and that eventually marked the relevant PR where we could track the entire to-do for `dev` branch as merged. I already reverted the push mistake and deleted the release+tag caused by that mistake, but reopening the PR is a must too...

**Comment by kurellajunior** [comment]
Also from our side, this happens more often than it should. Reopening a PR should simply be possible, if the branch still exists and is not merged. For sanity we often have to roll back, which indeed means to set back the main branch and restore the feature branch. Then adding something to the feature branch and redo the PR. In most cases it is sufficient to have a new PR with the old one being plain wrong (not merged anymor) in the history causing only mild confusion.
But sometimes it is important to continue on the PR where it was merged. Can you please consider reopening a PR if the branch exists and its HEAD does not have a merge successor?

**Comment by ragnarok22** [comment]
2025 and I ask for the same feature 😅

**Comment by bford** [comment]
Same here.  In my case not because of an accidental merge, but because a lot of work had been done in a still-incomplete branch and we decided the work-in-progress was stable enough that its progress so far should be merged into main even though the larger project represented by the branch and PR was still ongoing.  Not standard practice but there can be justifications.

In this case, the "accident" wasn't the merge itself, per se, but rather doing the merge with GitHub rather than at Git level.  I would have done the latter if I had known that there was no way to re-open a merged PR.  I was even careful *not* to delete the branch, and we didn't need or want to revert the Git commits since merging those commits was intentional.  We just would like to reopen the PR, still using the same branch, so that the conversation history and view status etc doesn't have to be lost in the transition to a new PR as discussed above.

So +1 on this feature request (I would say bug fix), please.

**Reply by kurtbahartr** [reply]
Doing merge with Git still marks the PR as merged so it's related to how GitHub interprets PRs rather than the web UI.

**Comment by reynadev-oss** [comment]
I’ve run into this too — a PR was merged before review was finished, and I lost the “viewed files” state and context for unresolved comments. Being able to reopen a merged PR (when the branch still exists) would make reviews much smoother.

**Comment by MusicalNinjaDad** [comment]
My usecase: new repo, PR1 with valuable comments and linked issues. Forgot I'd not yet restricted merge types and boom - fast-forward main, which I didn't want (don't troll/flame that point please!)
So I can:
 - reset --hard and push --force main
 - restore the deleted branch
 - correct the repo's PR settings (up to this point it's my fault ...)
 - open a new PR#7 and comment-link to PR#1 (I'd like to reopen PR#1 instead, happy for this to be cli only or require 2FA sudo ...)
 - but lose the traceability of the comments & issues, because I now have the actual merge from PR#7 and all the comments & links in PR#1 (bad UX IMV)
